Index: work/src/edu/arizona/kfs/module/ld/document/web/struts/YearEndSalaryExpenseTransferForm.java
===================================================================
--- work/src/edu/arizona/kfs/module/ld/document/web/struts/YearEndSalaryExpenseTransferForm.java	(revision 7229)
+++ work/src/edu/arizona/kfs/module/ld/document/web/struts/YearEndSalaryExpenseTransferForm.java	(revision 7230)
@@ -15,8 +15,7 @@
  */
 package edu.arizona.kfs.module.ld.document.web.struts;
 
-import edu.arizona.kfs.fp.businessobject.ErrorCertification;
-import edu.arizona.kfs.module.ld.document.YearEndSalaryExpenseTransferDocument;
+import org.kuali.kfs.module.ld.document.web.struts.SalaryExpenseTransferForm;
 
 public class YearEndSalaryExpenseTransferForm extends SalaryExpenseTransferForm {
     public YearEndSalaryExpenseTransferForm() {
